aixplain/modules/model/utils.py

```diff
@@ -0,0 +1,70 @@
+__author__ = "thiagocastroferreira"
+
+import json
+import logging
+from aixplain.utils.file_utils import _request_with_retry
+from typing import Dict, Text, Union
+
+
+def build_payload(data: Union[Text, Dict], parameters: Dict = {}):
+    from aixplain.factories import FileFactory
+
+    data = FileFactory.to_link(data)
+    if isinstance(data, dict):
+        payload = data
+    else:
+        try:
+            payload = json.loads(data)
+            if isinstance(payload, dict) is False:
+                if isinstance(payload, int) is True or isinstance(payload, float) is True:
+                    payload = str(payload)
+                payload = {"data": payload}
+        except Exception:
+            payload = {"data": data}
+    payload.update(parameters)
+    payload = json.dumps(payload)
+    return payload
+
+
+def call_run_endpoint(url: Text, api_key: Text, payload: Dict) -> Dict:
+    headers = {"x-api-key": api_key, "Content-Type": "application/json"}
+
+    resp = "unspecified error"
+    try:
+        r = _request_with_retry("post", url, headers=headers, data=payload)
+        resp = r.json()
+    except Exception as e:
+        logging.error(f"Error in request: {e}")
+        response = {
+            "status": "FAILED",
+            "completed": True,
+            "error_message": "Model Run: An error occurred while processing your request.",
+        }
+
+    if 200 <= r.status_code < 300:
+        logging.info(f"Result of request: {r.status_code} - {resp}")
+        status = resp.get("status", "IN_PROGRESS")
```

Comment:
same here, what is status is not there? It should be but better be safe and handle the case it is not and fail gracefully

Reply:
If the status is not there, it assigns `IN_PROGRESS`. In any case, I did a modification to return a response with FAILED status when no `data` is returned.